crates/hirola-core/src/mixins.rs

```diff
@@ -163,3 +213,256 @@ pub mod model {
         Model(s.clone(), PhantomData)
     }
 }
+
+
+///Mixin for reactive classes
+/// 
+/// ## Example
+/// ```
+/// pub fn yellow_or_black_box(_app: &HirolaApp) -> Dom {
+///     let state = Signal::new(vec!["black-box".to_string()]);
+///
+///     let change_class = state.callback(move |st, _e:Event| {
+///         match st.get().to_vec()[0].as_str() {
+///             "black-box" => st.set(vec!["yellow-box".to_string()]),
+///             "yellow-box" => st.set(vec!["black-box".to_string()]),
+///             _ => panic!()
+///         }
+///     });
+/// 
+///     html! {
+///         <div class="" mixin:class=&class(&state)>
+///             <style>r##"
+///             .black-box {
+///                 background-color: black;
+///                 color: white;
+///             }
+///             .yellow-box {
+///                 background-color: yellow;
+///             }
+///             "##</style>
+///             <button on:click=change_class>"Click to change colour"</button>
+///         </div>
+///     }
+/// }
+/// ```
+pub fn class<'a>(signal: &'a Signal<Vec<String>>) -> Box<dyn Fn(DomNode) -> () + 'a> {
```

Comment:
This also makes good sense, but am curious if a macro would make more sense.

Reply:
Interesting, can I know where do you think we can call this macro from?

Reply:
https://github.com/geofmureithi/hirola/pull/6/files/e165ed0641d259ff12b4e86f58095072453668d4#r998411264